src/NuGet.Core/NuGet.Build.Tasks.Console/Program.cs

```diff
@@ -62,23 +62,30 @@ public static async Task<int> Main(string[] args)
             MSBuildFeatureFlags.EnableCacheFileEnumerations = true;
             MSBuildFeatureFlags.LoadAllFilesAsReadonly = true;
             MSBuildFeatureFlags.SkipEagerWildcardEvaluations = true;
-
-            // Only wire up an AssemblyResolve event handler if being debugged or running under a unit test
-            if (debug || string.Equals(Environment.GetEnvironmentVariable("UNIT_TEST_RESTORE_TASK"), bool.TrueString, StringComparison.OrdinalIgnoreCase))
+#if NETFRAMEWORK
+            if (AppDomain.CurrentDomain.Id == 1)
```

Comment:
```suggestion
            if (AppDomain.CurrentDomain.IsDefaultAppDomain())
```
This might be better.

Reply:
Thanks, I'll update this!